java/code/src/com/redhat/rhn/frontend/xmlrpc/recurringaction/RecurringActionHandler.java

```diff
@@ -0,0 +1,243 @@
+     *
+     * @xmlrpc.doc Return a list of recurring actions for a given entity.
+     * @xmlrpc.param #session_key()
+     * @xmlrpc.param #param_desc("string", "entityType", "Type of the target entity")
+     * @xmlrpc.param #param_desc("int", "entityId", "Id of the target entity")
+     * @xmlrpc.returntype
+     *      #array()
+     *          $RecurringActionSerializer
+     *      #array_end()
+     */
+    public List<? extends RecurringAction> listByEntity(User loggedInUser, String entityType, Integer entityId) {
+        List<? extends RecurringAction> schedules;
+        try {
+            switch (getEntityType(entityType)) {
+                case MINION:
+                    schedules = RecurringActionManager.listMinionRecurringActions(entityId, loggedInUser);
+                    break;
+                case GROUP:
+                    schedules = RecurringActionManager.listGroupRecurringActions(entityId, loggedInUser);
+                    break;
+                case ORG:
+                    schedules = RecurringActionManager.listOrgRecurringActions(entityId, loggedInUser);
+                    break;
+                default:
+                    throw new IllegalStateException("Unsupported type " + entityType);
+            }
+            return schedules;
+        }
+        catch (PermissionException e) {
+            throw new PermissionCheckFailureException(e.getMessage());
+        }
+    }
+
+    /**
+     * Return recurring action with given action id.
+     *
+     * @param loggedInUser The current user
+     * @param actionId id of the action
+     * @return the list of recurring actions
+     *
+     * @xmlrpc.doc Return recurring action with given action id.
+     * @xmlrpc.param #session_key()
+     * @xmlrpc.param #param_desc("int", "action_id", "Id of the action")
+     * @xmlrpc.returntype $RecurringActionSerializer
+     */
+    public RecurringAction lookupById(User loggedInUser, Integer actionId) {
+        try {
+            RecurringAction action = RecurringActionFactory.lookupById(actionId).orElseThrow();
+            if (action.canAccess(loggedInUser)) {
+                return action;
+            }
+        }
+        catch (NoSuchElementException e) {
+            throw new EntityNotExistsFaultException("Action with id: " + actionId + " does not exist");
+        }
+        throw new PermissionCheckFailureException("Action not accessible to user: " + loggedInUser);
+    }
+
+    /**
+     * Create a new recurring action.
+     *
+     * @param loggedInUser The current user
+     * @param actionProps Map containing action properties
+     * @return action id or exception thrown otherwise
+     *
+     * @xmlrpc.doc Create a new recurring action.
+     * @xmlrpc.param #session_key()
+     * @xmlrpc.param
+     *  #struct("actionProps")
+     *      #prop_desc("string", "entity_type", "The type of the target entity")
+     *      #prop_desc("string", "entity_id", "The id of the target entity")
```

Comment:
`entity_id` should be passed as `"int"`. This'd be better since the serializer returns the entity IDs as numbers (so that data returned from the `list`/`lookup` can be reused for creating/updating actions.).